modules/c++/sys/source/OSWin32.cpp

```diff
@@ -221,14 +222,21 @@ std::string sys::OSWin32::getEnv(const std::string& s) const
 
     // If we can use a normal size buffer, lets not bother to malloc
 
-    char *buffer = new char[size + 1];
-    DWORD retVal = GetEnvironmentVariable(s.c_str(), buffer, size);
-    result = buffer;
-    delete [] buffer;
-
+    std::vector<char> buffer;
```

Comment:
Just do this in one shot: `std::vector<char> buffer(size + 1);`
(Also note spaces between the operators there)